### PR TITLE
Add accepted values to the HtmlEditor.toolbar.items property (T810025)

### PR DESCRIPTION
**Pull request by mrjohn0011**

### CODE REVIEW

**Review by San4es — APPROVED** [review]

File: js/ui/html_editor/ui.html_editor.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ const HtmlEditor = Editor.inherit({
             */
             /**
             * @name dxHtmlEditorToolbar.items
-            * @type Array<dxHtmlEditorToolbarItem,string>
+            * @type Array<dxHtmlEditorToolbarItem,Enums.HtmlEditorToolbarItem>
```
Comment (San4es) [inline js/ui/html_editor/ui.html_editor.js:120]: `Array<dxHtmlEditorToolbarItem,string,Enums.HtmlEditorToolbarItem>`?

**Review by RomanTsukanov — COMMENTED** [review]
How about formatName? It misses accepted values as well

**Review by San4es — CHANGES_REQUESTED** [review]
Also we need to use the added enum for dxHtmlEditorToolbarItem.formatName

File: js/ui/html_editor/ui.html_editor.js (hunk 1 above)
Comment (RomanTsukanov) [reply js/ui/html_editor/ui.html_editor.js:120]: We need `string` only if custom strings are accepted. AFAIK, it isn't so.

**Comment by RomanTsukanov** [comment]
> 
> 
> Also we need to use the added enum for dxHtmlEditorToolbarItem.formatName

formatName accepts a different set of values and needs a separate enum

File: js/docEnums.js [hunk 2]
```diff
@@ -986,6 +986,16 @@
  * @enum {'html'|'markdown'}
  */
 
+/**
+ * @typedef {string} Enums.HtmlEditorToolbarItem
+ * @enum {'background'|'bold'|'color'|'italic'|'link'|'image'|'strike'|'subscript'|'superscript'|'underline'|'blockquote'|'header'|'increaseIndent'|'decreaseIndent'|'orderedList'|'bulletList'|'alignLeft'|'alignCenter'|'alignRight'|'alignJustify'|'codeBlock'|'variable'|'separator'|'undo'|'redo'|'clear'}
+ */
+
+/**
+ * @typedef {string} Enums.HtmlEditorToolbarFormatName
```
Comment (RomanTsukanov) [inline js/docEnums.js]:
```suggestion
 * @typedef {string} Enums.HtmlEditorFormat
```

File: js/ui/html_editor/ui.html_editor.js [hunk 3]
```diff
@@ -126,7 +126,7 @@ const HtmlEditor = Editor.inherit({
             */
             /**
             * @name dxHtmlEditorToolbarItem.formatName
-            * @type string
+            * @type Enums.HtmlEditorToolbarFormatName
```
Comment (RomanTsukanov) [inline js/ui/html_editor/ui.html_editor.js]:
```suggestion
            * @type Enums.HtmlEditorFormat
```

**Review by RomanTsukanov — DISMISSED** [review]

**Review by San4es — APPROVED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**